snuba/query/validation/validators.py

```diff
@@ -147,7 +147,16 @@ class SubscriptionAllowedClausesValidator(QueryValidator):
     def __init__(self, max_allowed_aggregations: int) -> None:
         self.max_allowed_aggregations = max_allowed_aggregations
 
-    def validate(self, query: Query, alias: Optional[str] = None) -> None:
+    def validate(
+        self,
+        query: Query,
+        alias: Optional[str] = None,
+        disallowed: Optional[Sequence[str]] = None,
```

Comment:
Does this actually type check ?
The validate method comes from the parent class 
`def validate(self, query: Query, alias: Optional[str] = None,) -> None:` 
Adding a parameter violates the parent interface.

If you want to provide a parameter used by the implementation but remaining compatible with the parent class interface, please pass the list of disallowed aggregation to the constructor, store it as an attribute and use it in the validate method.
Also please give the parameter a name that reflects more accurately what this argument is for (like `disallowed_aggregations`).